barretenberg/cpp/src/barretenberg/crypto/pedersen_hash/c_bind.cpp

```diff
@@ -7,44 +7,16 @@ extern "C" {
 
 WASM_EXPORT void pedersen_hash__init() {}
 
-WASM_EXPORT void pedersen__hash(uint8_t const* inputs_buffer, uint8_t* output)
+WASM_EXPORT void pedersen__hash_with_hash_index(uint8_t const* inputs_buffer, uint32_t hash_index, uint8_t* output)
 {
     std::vector<grumpkin::fq> to_compress;
     read(inputs_buffer, to_compress);
-    auto r = crypto::pedersen_hash::hash(to_compress);
+    crypto::GeneratorContext<curve::Grumpkin> ctx; // todo fix
+    ctx.offset = static_cast<size_t>(hash_index);
+    auto r = crypto::pedersen_hash::hash(to_compress, ctx);
     barretenberg::fr::serialize_to_buffer(r, output);
 }
```

Comment:
The previous `pedersen__hash_with_hash_index` was using `uint32_t const* hash_index` whereas the previous `pedersen__compress_with_hash_index was using `uint32_t const hash_index` -- I set this to match the compress variant, though note its now different to the c_binds because they are using `uint32_t const*`